examples/src/main/scala/za/co/absa/fadb/examples/enceladus/DatasetSchema.scala

```diff
@@ -60,72 +59,63 @@ object DatasetSchema {
+    with UserDefinedStatusHandling {
 
-
-    override protected def sqlToCallFunction(values: SchemaInput): SQLActionBuilder = {
+    override protected def sql(values: SchemaInput): SQLActionBuilder = {
       sql"""SELECT A.status, A.status_text, A.id_schema_version
-             FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
-                ${values.fields}::JSONB, ${values.userName}
-             ) A;"""
+            FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
+              ${values.fields}::JSONB, ${values.userName}
+            ) A;"""
     }
 
-    override protected def resultConverter: GetResult[Long] = {
-      val gr:GetResult[Long] = GetResult(r => {
-        val status: Int = r.<<
-        val statusText: String = r.<<
-        if (status != 201) throw new StatusException(status, statusText)
-        r.<<
-      })
-      gr
-    }
+    override protected def slickConverter: GetResult[Long] = GetResult.GetLong
+
+    override def OKStatuses: Set[Integer] = Set(201)
+
   }
 
-  final class GetSchema(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBUniqueFunction[ExecutorEngineType, (String, Option[Int]), Schema](schema)
-    with SlickPgFunction[(String, Option[Int]), Schema] {
+  final class GetSchema(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBUniqueFunction[(String, Option[Int]), Schema, SlickPgEngine]
+    with SlickPgFunctionWithStatusSupport[(String, Option[Int]), Schema]
+    with UserDefinedStatusHandling {
 
     def apply(id: Long): Future[Schema] = {
       val sql =
         sql"""SELECT A.*
              FROM #$functionName($id) A;"""
 
-      schema.unique(makeQueryFunction(sql)(resultConverter))
+      val slickQuery = new dBEngine.QueryType(sql, slickConverter)
+      dBEngine.unique[Schema](slickQuery)
     }
 
-    override protected def sqlToCallFunction(values: (String, Option[Int])): SQLActionBuilder = {
+    override protected def sql(values: (String, Option[Int])): SQLActionBuilder = {
       sql"""SELECT A.*
-              FROM #$functionName(${values._1}, ${values._2}) A;"""
+            FROM #$functionName(${values._1}, ${values._2}) A;"""
+    }
+
+    override protected val slickConverter: GetResult[Schema] = GetResult{r =>
+      Schema(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
     }
 
-    override protected def resultConverter: GetResult[Schema] = DatasetSchema.GetSchemaImplicit
+    override val OKStatuses: Set[Integer] = Set(200)
   }
 
-  final class ListSchemas(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBSeqFunction[ExecutorEngineType, Boolean, SchemaHeader](schema)
+  final class List(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBSeqFunction[Boolean, SchemaHeader, SlickPgEngine]()
```

Comment:
Why the status support/handling were not used here? What's different there? I think that 'list' operation is similar to 'get' i.e. the DB function should return some status code and status text for us?

Reply:
`list` operations returning a number of results and being strictly read-only, don't have to have status results. I still prefer that, while I think @Zejnilovic less.
Anyway, here it works again as an example - _status_ is not mandatory.
In _Unify_ we can (and should) still discuss this concept ❗